src/main/java/lottogame/domain/LottoNumbers.java

```diff
@@ -21,6 +24,17 @@ public List<LottoNumber> getLottoNumbers() {
         return lottoNumbers;
     }
 
+    public int countMatchingNumbers(LottoNumbers other) {
+        return (int) this.lottoNumbers.stream()
+                .filter(other.lottoNumbers::contains)
+                .count();
+    }
+
+    public boolean isMatchingBonus(LottoNumber bonusNumber) {
+        return this.lottoNumbers.stream()
+                .anyMatch(number -> number.equals(bonusNumber));
```

Comment:
List의 `contains()`를 활용해봐도 좋을 것 같습니다.
```suggestion
        return this.lottoNumbers.contains(bonusNumber);
```